docs/performance/productizing.rst

```diff
@@ -0,0 +1,116 @@
+.. _topics-to-watch-out-when-productizing-cratedb:
+
+###############################################
+ Topics to watch out when productizing CrateDB
+###############################################
+
+.. _mindful-of-memory:
+
+*******************************
+ Be mindful of memory capacity
+*******************************
+
+In CrateDB, operations requiring a working set like groupings, aggregations, and
+sorting are performed fully in memory without spilling over to disk.
+
+Sometimes you may have a query that leads to a sub-optimal execution plan
+requiring lots of memory. If you are coming to CrateDB from other database
+systems, your experience may be that these queries will proceed to run taking
+longer than required and impacting other workloads in the meanwhile. Sometimes
+this effect may be obvious if a query takes a lot of resources and runs for a
+long time, other times it may go unnoticed if a query that could complete in say
+100 milliseconds takes one hundred times longer, 10 seconds, but the users put
+up with it without reporting to you.
+
+In CrateDB if a query would require more heap memory than the interested nodes
+have available the query will fail with a particular type of error message that
+we call a `CircuitBreakerException`, this is a fail-fast approach, we can see
+quickly that there is an issue and optimize the query to get the best
+performance, without impacting other workloads.
+
+Please take a look at :ref:`Query Optimization 101 <performance-optimization>` for strategies to optimize your
+queries when you encounter this situation.
+
+.. _reading-lots-of-records:
+
+*************************
+ Reading lots of records
+*************************
+
+When the HTTP endpoint is used CrateDB will prepare the entire response in
+memory before sending it to the client.
+
+When the PostgreSQL protocol is used CrateDB attempts to stream the results but
+in many cases it still needs to bring all rows to the query handler node first.
+
+So we should always limit how many rows we request at a time, see `Fetching
+large result sets from CrateDB`_.
+
+.. _number-of=shards:
+
+******************
+ Number of shards
+******************
+
+In CrateDB data in tables and partitions is distributed in storage units that we
+call shards.
+
+If we do not specify how many shards we want for a table/partition CrateDB will
+derive a default from the number of nodes.
```

Comment:
The default is **4**, it will **not** derive it from the number of nodes.

Reply:
https://cratedb.com/docs/crate/reference/en/latest/sql/statements/create-table.html#clustered
says:

> If not provided, the number of shards is calculated based on the number of currently active data nodes

Reply:
Oh right, my fault! 🤦  I should probably develop more often with > 4 nodes ;)